script/ide/System/Core/Color.lua

```diff
@@ -224,4 +224,28 @@ function Color.ChangeOpacity(color, opacity)
 		color = Color.ToValue(color);
 	end
 	return (opacity or 255)*alphaMask + band(color, 0xffffff)
+end
+
+-- @param color1 and color2: string like "#ff0000"
+function Color.Multiply(color1, color2)
+	local color1_num = {};
```

Comment:
move local color_XXX = {} to file scope, avoiding creating additional table for each invocation